src/deadline/job_attachments/api/attachment.py

```diff
@@ -140,17 +140,34 @@ def attachment_upload(
                 f"No valid root defined for given manifest {file_name}, please check input root dirs and path mapping rule."
             )
 
+        metadata = {"Metadata": {"asset-root": json.dumps(rule.source_path, ensure_ascii=True)}}
+        # S3 metadata must be ASCII, so use either 'asset-root' or 'asset-root-json' depending
+        # on whether the value is ASCII.
+        try:
+            # Add the 'asset-root' metadata if the path is ASCII
+            rule.source_path.encode(encoding="ascii")
```

Comment:
nice catch! We totally buried some hidden details

Reply:
That's a good question, it comes from here https://github.com/aws-deadline/deadline-cloud/pull/492